### DISCUSSION TITLE
Add DynamoDB as a Bundled Store for Sessions

### DISCUSSION THREAD

**Discussion by 7nohe** [adonisjs/core#4710] [Ideas]
I’d like to propose adding DynamoDB as a bundled store option for session management. DynamoDB is widely used, especially in AWS environments, and it could be a great addition for developers who are already leveraging AWS services. 

Laravel has already included support for DynamoDB as a session store.
https://laravel.com/docs/11.x/session#configuration

If this idea aligns with the project’s direction, I’d be happy to help with the implementation or collaborate with others who might be interested. Looking forward to hearing your thoughts!

**Comment by thetutlage** [comment]
@7nohe Sounds good. Feel free to send a PR to add DynamoDB support. One thing you have to see is how to run tests against a real DynamoDB server. 

It will be nice if we have something to use without creating an AWS account. But yeah, let me know if that isn't possible

**Reply by 7nohe** [reply]
I would definitely like to give it a try. 
For testing and running locally, we can use DynamoDB Local, which allows you to operate without an AWS account.

**Reply by 7nohe** [reply]
@thetutlage 
I've created a [PR](https://github.com/adonisjs/session/pull/87).
I've also made all possible preparations for the real DynamoDB server. It seems to be working well on my own AWS account.

**Comment by 7nohe** [comment]
DynamoDB is now supported in v7.5.0.

https://github.com/adonisjs/session/releases/tag/v7.5.0
